builder/grpc/rust/compile.bzl

```diff
@@ -0,0 +1,62 @@
+#
```

Comment:
The rule implementation is quite straightforward:

1. Define the inputs for the compile script, namely all `.proto` sources and the `protoc` executable.
2. Define the outputs of the compile script. One output is generated for each unique `package` defined in our `.proto` files. Bazel can't (easily!) get these package names, so the user is required to supply them as a parameter.
3. Run the compile script and expose the outputs as source files that any `rust_library` can depend on.

(The `rust_library` is then expected to depend on the `tonic` and `prost` crates, otherwise it will fail to compile. This is in line with Java, where we expect our `java_library` to depend on the correct gRPC + Protobuf JARs.)